Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -51,19 +81,290 @@ namespace AZ::SceneAPI::Behaviors
+            graph,
+            graph.GetRoot(),
+            graph.GetContentStorage().cbegin(),
+            true);
+
+        if (view.begin() == view.end())
+        {
+            return Events::ProcessingResult::Ignored;
+        }
+        for (auto it = view.begin(); it != view.end(); ++it)
+        {
+            Containers::SceneGraph::NodeIndex currentIndex = graph.ConvertToNodeIndex(it.GetHierarchyIterator());
+            AZStd::string currentNodeName = graph.GetNodeName(currentIndex).GetPath();
+            auto currentContent = graph.GetNodeContent(currentIndex);
+            if (currentContent)
+            {
+                if (azrtti_istypeof<AZ::SceneAPI::DataTypes::ITransform>(currentContent.get()))
+                {
+                    auto parentIndex = graph.GetNodeParent(currentIndex);
+                    if (parentIndex.IsValid() == false)
+                    {
+                        continue;
+                    }
+                    auto parentContent = graph.GetNodeContent(parentIndex);
+                    if (parentContent && azrtti_istypeof<AZ::SceneAPI::DataTypes::IMeshData>(parentContent.get()))
+                    {
+                        // map the node parent to the ITransform
+                        MeshTransformPair pair {parentIndex, currentIndex};
+                        meshTransformMap.emplace(MeshTransformEntry{ graph.GetNodeParent(parentIndex), AZStd::move(pair) });
+                    }
+                }
+            }
+        }
+
+        if (meshTransformMap.empty())
+        {
+            return Events::ProcessingResult::Ignored;
+        }
+
+        // compute the filenames of the scene file
+        AZStd::string watchFolder = scene.GetWatchFolder() + "/";
+        AZStd::string relativeSourcePath = scene.GetSourceFilename();
+        AZ::StringFunc::Replace(relativeSourcePath, ".", "_");
+        AZ::StringFunc::Replace(relativeSourcePath, watchFolder.c_str(), "");
+        AZStd::string filenameOnly{ relativeSourcePath };
+        AZ::StringFunc::Path::GetFileName(filenameOnly.c_str(), filenameOnly);
+        AZ::StringFunc::Path::ReplaceExtension(filenameOnly, "prefab");
+
+        // create mesh groups and Editor entities for each pair
+        AZStd::vector<AZStd::shared_ptr<DataTypes::IManifestObject>> manifestUpdates;
+        AZStd::unordered_map<Containers::SceneGraph::NodeIndex, AZ::EntityId> nodeEntityMap;
+
+        for (const auto& entry : meshTransformMap)
+        {
+            const auto thisNodeIndex = entry.first;
+            const auto meshNodeIndex = entry.second.first;
+            const auto meshNodeName = graph.GetNodeName(meshNodeIndex);
+            AZStd::string meshNodePath { meshNodeName.GetPath() };
+
+            AZStd::string meshNodeFullName;
+            meshNodeFullName = relativeSourcePath;
+            meshNodeFullName.append("_");
+            meshNodeFullName.append(meshNodeName.GetName());
```

Comment:
do you need to check for uniqueness at all? It's probably unlikely, and it might be automatically fixed upstream, but I figured I would ask just in case.

Reply:
in this case we are okay since the default mesh group name is determined differently, and these are the "default mesh names" when there is no scene manifest